lib/service/invoicesubscription.go

```diff
@@ -236,14 +243,16 @@ func (svc *LndhubService) InvoiceUpdateSubscription(ctx context.Context) error {
 	for {
 		select {
 		case <-ctx.Done():
-			return fmt.Errorf("Context was canceled")
+			return context.Canceled
 		default:
 			// receive the next invoice update
 			rawInvoice, err := invoiceSubscriptionStream.Recv()
+			// in case of an error, we want to return and restart LNDhub
+			// in order to try and reconnect the gRPC subscription
 			if err != nil {
 				svc.Logger.Errorf("Error processing invoice update subscription: %v", err)
 				sentry.CaptureException(err)
-				continue
+				return err
```

Comment:
where will this error end up?

Reply:
[here](https://github.com/getAlby/lndhub.go/blob/catch-add-index-error/main.go#L184)

Reply:
ah because it is returning, otherwise it would loop and not reach there, ok. thanks.